root/frontend/src/pages/HistoryPage.jsx

```diff
@@ -1,13 +1,409 @@
+         const link = document.createElement("a");
+         link.download = `${fileName}.json`;
+         const blob = new Blob([JSON.stringify({"platform":"wewe",
+         "created_at":"asas", 
+        "post_id":"efe",
+         "lang":"th",
+        "title":"rtr",
+         "text":"dsdc", 
+        "author_id":"asa",
+        "positive_votes":3,
+         "has_embedded_media":"xzx",
+         "comment_count":9, 
+        "top_comments":"dda",
+         "sentiment_score":5,
+         "sentiment_confidence":4
+        })], { type: "json" });
+         link.href = URL.createObjectURL(blob);
+         link.click();
+         URL.revokeObjectURL(link.href);
+     
+        
+       };
+       render() {
+         return (
+           <div className="App">
+             <button onClick={this.handleDownloadDoc("resultats")}>
+               Download
+             </button>
+           </div>
+         );
+       }
+     }
+     
+     const rootElement = document.getElementById("root");
+     ReactDOM.render(<App />, rootElement);
+     */
+ /*import createHistory from "history/createBrowserHistory";
+
+const history = createHistory();
+
+export default class Page extends React.Component {
+  state = {
+    value: "",
+    query: "",
+    data: ["foo", "bar", "test"],
+    filteredData: []
+  };
+
+  componentDidMount() {
+    this.setQuery();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.query !== this.props.match.params.query) {
+      this.setQuery();
+    }
+  }
+
+  setQuery = () => {
+    const { query = "" } = this.props.match.params;
+    const filteredData = this.state.data.filter(element =>
+      element.toLowerCase().includes(query.toLowerCase())
+    );
+    this.setState({ value: query, query, filteredData });
+  };
+
+  onChange = event => {
+    this.setState({ value: event.target.value });
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+    history.push(`/${this.state.value}`);
+  };
+
+  render() {
+    const { value, filteredData } = this.state;
+
+    return (
+      <div>
+        <form onSubmit={this.onSubmit}>
+          <input value={value} onChange={this.onChange} />
+          <input type="submit" value="Search" />
+          {filteredData.map(element => <div key={element}> {element} </div>)}
+        </form>
+      </div>
+    );
+  }
+}
+
+class App extends React.Component {
+  render() {
+    return (
+      <Router history={history}>
+        <Switch>
+          <Route path="/:query?" component={Page} />
+        </Switch>
+      </Router>
+    );
+  }
+}
+  */   
+/*import React from "react";
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Link,
+  useNavigate 
+} from "react-router-dom";
+
+const Foo = () => {
+  const navigate = useNavigate();
+
   return (
-    <ContentContainer>
-      <div>history page</div>
-    </ContentContainer>
+    <div>
+      <button onClick={navigate.goBack}>Back</button>
+      <p>foo</p>
+    </div>
```

Comment:
```suggestion
    <ContentContainer>
      <div>
        <button onClick={navigate.goBack}>Back</button>
        <p>foo</p>
      </div>
    </ContentContainer>
```

You should keep this page wrapper so that it has similar formatting to the other pages instead of a blank space.